src/Tes.RunnerCLI/Commands/NodeTaskUtils.cs

```diff
@@ -31,6 +42,55 @@ public static async Task<NodeTask> DeserializeNodeTaskAsync(string tesNodeTaskFi
         }
     }
 
+    public async Task SerializeNodeTaskAsync(NodeTask nodeTask, string tesNodeTaskFilePath)
+    {
+        try
+        {
+            var nodeTaskText = JsonSerializer.Serialize(nodeTask, jsonSerializerOptions) ?? throw new InvalidOperationException("The JSON data provided is invalid.");
+
+            await File.WriteAllTextAsync(tesNodeTaskFilePath, nodeTaskText);
+        }
+        catch (Exception e)
+        {
+            Logger.LogError(e, "Failed to serialize task JSON file.");
+            throw;
+        }
+    }
+
+    public async Task<NodeTask> ResolveNodeTaskAsync(FileInfo? file, Uri? uri, Lazy<NodeTaskResolverOptions> options)
```

Comment:
Tests for this functionality are missing, I understand that part of the challenge is that HttpBlobApiUtils is static, but you can create a wrapper that can be mocked very easily.